Source/Tabs/DataProviders/Cancelling.lua

```diff
@@ -2,70 +2,125 @@ local CANCELLING_TABLE_LAYOUT = {
   },
-  {
-    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
-    headerText = "",
-    cellTemplate = "AuctionatorDeleteButtonCellTemplate",
-    width = 40,
-  }
 }
 
-CancellingDataProviderMixin = CreateFromMixins(DataProviderMixin)
+local DATA_EVENTS = {
+  "OWNED_AUCTIONS_UPDATED",
+  "AUCTION_CANCELED"
+}
+
+AuctionatorCancellingDataProviderMixin = CreateFromMixins(DataProviderMixin, AuctionatorItemKeyLoadingMixin)
 
-function CancellingDataProviderMixin:OnLoad()
+function AuctionatorCancellingDataProviderMixin:OnLoad()
   DataProviderMixin.OnLoad(self)
+  AuctionatorItemKeyLoadingMixin.OnLoad(self)
+  Auctionator.EventBus:Register(self, {Auctionator.Cancelling.Events.RequestCancel})
+
+  self.waitingforCancellation = {}
+  self.beenCancelled = {}
+
+end
+
+function AuctionatorCancellingDataProviderMixin:OnShow()
+  C_AuctionHouse.QueryOwnedAuctions({})
+
+  FrameUtil.RegisterFrameForEvents(self, DATA_EVENTS)
+end
+
+function AuctionatorCancellingDataProviderMixin:OnHide()
+  FrameUtil.UnregisterFrameForEvents(self, DATA_EVENTS)
+end
+
+local COMPARATORS = {
+  price = Auctionator.Utilities.NumberComparator,
+  name = Auctionator.Utilities.StringComparator,
+  quantity = Auctionator.Utilities.NumberComparator
+}
+
+function AuctionatorCancellingDataProviderMixin:Sort(fieldName, sortDirection)
+  local comparator = COMPARATORS[fieldName](sortDirection, fieldName)
+
+  table.sort(self.results, function(left, right)
+    return comparator(left, right)
+  end)
+
+  self.onUpdate(self.results)
+end
+
+function AuctionatorCancellingDataProviderMixin:OnEvent(eventName, auctionID, ...)
+  AuctionatorItemKeyLoadingMixin.OnEvent(self, event, ...)
+  if eventName == "AUCTION_CANCELED" then
+    table.insert(self.beenCancelled, auctionID)
+    C_AuctionHouse.QueryOwnedAuctions({})
+
+  elseif eventName == "OWNED_AUCTIONS_UPDATED" then
+    self:Reset()
+    self:PopulateAuctions()
+  end
+end
+
+function AuctionatorCancellingDataProviderMixin:ReceiveEvent(eventName, eventData, ...)
+  if eventName == Auctionator.Cancelling.Events.RequestCancel then
+    table.insert(self.waitingforCancellation, eventData)
+  end
+end
+
+function AuctionatorCancellingDataProviderMixin:PopulateAuctions()
+  local results = {}
+  for index = 1, C_AuctionHouse.GetNumOwnedAuctions() do
+    local info = C_AuctionHouse.GetOwnedAuctionInfo(index)
 
-  -- Example entry (note that the field names can change but you need to update the layout, above)
-  table.insert(
-    self.results,
-    {
-      itemKey = {
-        itemLevel = 46,
-        itemSuffix = 0,
-        itemID = 10506,
-        battlePetSpeciesID = 0
-      },
-      totalQuantity = 5,
-      price = 4207600,
-      duration = 24,
-      itemName = "Fancy item name string",
-      iconTexture = 133151
-    }
-  )
+    --Only look at unsold and uncancelled (yet) auctions
+    if info.status == 0 and
+       Auctionator.Utilities.ArrayIndex(self.beenCancelled, info.auctionID) == nil then
```

Comment:
Something off with alignment here... maybe use parentheses to make it clear that the entire thing is the if guard?
```lua
if (
  info.status == 0 and
  Auctionator.Utilities.ArrayIndex(self.beenCancelled, info.auctionID) == nil
) then
```
(Might even work without parens...) Could also just refactor the guard into a function:
```lua
if self:AuctionIsValid(info) then
```